src/gululog_idx.erl

```diff
@@ -234,6 +235,35 @@ delete_from_cache(Tid, LogId) ->
   end,
   Tid.
 
+%% @doc Truncate after given logid from cache and index file
+%% Return new index and index file for deleted or truncated
+%% @End
+-spec truncate(dirname(), index(), segid(), logid(), ?undef | dirname()) ->
+        {index(), [filename()]}.
+truncate(Dir, #idx{tid = Tid, fd = Fd} = Idx, SegId, LogId, BackupDir) ->
+  Ms = ets:fun2ms(fun(?ETS_ENTRY(_, LogIdX, _) = EtsEntry) when LogIdX >= LogId -> EtsEntry end),
+  [_ | _] = EtsEntryList = ets:select(Tid, Ms),
+  lists:map(fun(?ETS_ENTRY(_, LogIdX, _)) -> ets:delete(Tid, LogIdX) end, EtsEntryList),
+  DeleteList = lists:usort(lists:filtermap(fun(?ETS_ENTRY(SegIdX, _, _)) ->
+                                             SegIdX > SegId
```

Comment:
the advantage is that you dont have to create a intermediate list comparing to map + filter or filter + map.